<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xmlns:context="http://www.springframework.org/schema/context"
	xmlns:p="http://www.springframework.org/schema/p"
	xmlns:util="http://www.springframework.org/schema/util"
	xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
		http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context-4.3.xsd
		http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util-4.3.xsd">

	
	<!-- it will do exception  -->
	
	<!-- <bean class="com.app.model.Address" name="address"> 
		<constructor-arg name="emp" ref="employee"/>
	</bean>
	
	<bean class="com.app.model.Employee" name="employee">
			<constructor-arg name="addr" ref="address" />    for address class
	 </bean>
	  -->
	 <!-- to overcome this problem -->
	 
	 <bean class="com.app.model.Address" name="address"> 
		<property name="emp" ref="employee"/>
	</bean>
	
	<bean class="com.app.model.Employee" name="employee">
			<property name="addr" ref="address" />    <!-- for address class -->
	 </bean>
	 
	 
	 
	</beans>
